simphony/io/file_mesh.py

```diff
@@ -0,0 +1,763 @@
+
+    Provides the column definition to store Points
+    forming a cell and its data.
+
+    """
+
+    uuid = tables.StringCol(32, pos=0)
+    data = tables.StringCol(32, pos=1)
+    points_uuids = tables.StringCol(
+        32, pos=2, shape=(MAX_POINTS_IN_CELL,)
+        )
+    n_points = tables.UInt32Col(pos=3)
+
+
+class _MeshDescriptor(tables.IsDescription):
+    """ Descriptor for storing Mesh information
+
+    Provides the column definition to store mesh data.
+
+    """
+
+    uuid = tables.StringCol(32, pos=0)
+
+
+class FileMesh(object):
+    """ FileMesh.
+
+    Interface of the mesh file driver.
+    Stores general mesh information Points and Elements
+    such as Edges, Faces and Cells and provide the
+    methods to interact with them. The methods are
+    divided in four diferent blocks:
+
+    (1) methods to get the related item with the provided uuid;
+    (2) methods to add a new item or replace;
+    (3) generator methods that return iterators
+        over all or some of the mesh items and;
+    (4) inspection methods to identify if there are any edges,
+        faces or cells described in the mesh.
+
+    Attributes
+    ----------
+    data : Data
+        The X coordinate.
+    points : dictionary of Point
+        Points of the mesh.
+    edges : dictionary of Edge
+        Edges of the mesh.
+    faces : dictionary of Face
+        Faces of the mesh.
+    cells : dictionary of Cell
+        Cells of the mesh.
+
+    See Also
+    --------
+    get_point, get_edge, get_face, get_cell
+    add_point, add_edge, add_face, add_cell
+    update_point, update_edge, update_face, update_cell
+    iter_points, iter_edges, iter_faces, iter_cells
+    has_edges, has_faces, has_cells
+
+    """
+
+    def __init__(self, group, meshFile):
+
+        self._file = meshFile
+        self._group = group
+        self._create_data_table()
+
+        if "points" not in self._group:
+            self._create_points_table()
+
+        if "edges" not in self._group:
+            self._create_edges_table()
+
+        if "faces" not in self._group:
+            self._create_faces_table()
+
+        if "cells" not in self._group:
+            self._create_cells_table()
+
+    def get_point(self, p_uuid):
+        """ Returns a point with a given uuid.
+
+        Returns the point stored in the mesh
+        identified by uuid. If such point do not
+        exists an exception is raised.
+
+        Parameters
+        ----------
+        uuid : int
+            uuid of the desired point.
+
+        Returns
+        -------
+        Point
+            Mesh point identified by uuid
+
+        Raises
+        ------
+        Exception
+            If the point identified by uuid was not found
+
+        """
+
+        for row in self._group.points.where(
+                'uuid == value', condvars={'value': p_uuid.hex}):
+            return Point(
+                tuple(row['coordinates']),
+                uuid.UUID(hex=row['uuid'], version=4),
```

Comment:
@itziakos, @nathanfranklin I'm having some problems using "bytes" in order to store the representation of the uuid, as you can see in the Travis report (https://travis-ci.org/simphony/simphony-common/builds/49300357). It randomly fails in some cases. Lets say 1 of every 200 tests ends up in a *ValueError: bytes is not a 16-char string* when trying to regenerate the uuid from the field stored

For now, it seems that storing the hex representation instead of the bytes values does the job. Have you any insight of what can be the reason of this behavior?